tests/test_vcs.py

```diff
@@ -88,6 +88,26 @@ def test_local_clone() -> None:
     shutil.rmtree(local_tmp, ignore_errors=True)
 
 
+@pytest.mark.impure
+def test_local_dirty_clone() -> None:
+    tmp = clone("https://github.com/copier-org/autopretty.git")
+    assert tmp
+    assert Path(tmp, "copier.yml").exists()
+
+    with local.cwd(tmp):
+        Path("unknown.txt").write_text("hello world")
+        Path("README.md").write_text("override string!!")
+
+    with pytest.warns(DirtyLocalWarning):
+        local_tmp = clone(tmp)
+
+    assert local_tmp
+    assert Path(local_tmp, "unknown.txt").exists()
+    assert Path(local_tmp, "README.md").exists()
+    assert Path(local_tmp, "copier.yaml").exists()
+    shutil.rmtree(local_tmp, ignore_errors=True)
+
+
```

Comment:
Instead of cloning a remote repo, which makes the test impure, could you build a local repo instead? For example something like this (untested):

```python
def test_local_dirty_clone(tmp_path_factory: pytest.TempPathFactory) -> None:
    src = map(tmp_path_factory.mktemp, ("src",))
    build_file_tree({src / "foo.txt": "bar"})
    git_save(src)
    build_file_tree({src / "foo.txt": "baz", src / "hello.txt": "world"})

    with pytest.warns(DirtyLocalWarning):
        tmp = clone(src)

    assert tmp
    assert Path(tmp, "foo.txt").exists()
    assert Path(tmp, "foo.txt").read_text() == "baz"
    assert Path(tmp, "hello.txt").exists()
    assert Path(tmp, "hello.txt").read_text() == "world"
```